core/src/main/java/com/sap/ai/sdk/core/AiCoreDestination.java

```diff
@@ -0,0 +1,28 @@
+package com.sap.ai.sdk.core;
+
+import com.sap.cloud.sdk.cloudplatform.connectivity.Destination;
+import com.sap.cloud.sdk.services.openapi.apiclient.ApiClient;
+import javax.annotation.Nonnull;
+
+/** Container for an API client and destination. */
+@FunctionalInterface
+public interface AiCoreDestination {
```

Comment:
I think we might be able to omit this interface, because we are refactoring orchestration to no longer need an `ApiClient`, and going forward that will probably hold true for any other generated code as well.

Thus, currently the only place we need the customized ApiClient is for the AI API, but for that we already have the convenience method, since its in the same maven module.

So maybe we can simplify. That also saves us from thinking about naming, because for me `AiCoreDestination` is too close to `Destination`, without actually implementing the interface.

Reply:
Currently in `main` branch

* method `destination()` is required:
  * explicitly by OpenAI integration
  * implicitly by `client()`
* method `client()` is required:
  * explicit by Orchestration and Core integration